crates/subspace-farmer/src/commitments/tests.rs

```diff
@@ -110,3 +110,78 @@ async fn find_by_tag() {
         );
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn partial_replica_commitments() {
+    init();
+    let base_directory = TempDir::new().unwrap();
+    let salt: Salt = [1u8; 8];
+
+    let max_pieces = 64;
+
+    let address = rand::random::<[u8; 32]>().into();
+    let plot = Plot::open_or_create(
+        &base_directory,
+        address,
+        Some(max_pieces * PIECE_SIZE as u64),
+    )
+    .unwrap();
+    let commitments = Commitments::new(base_directory.path().join("commitments")).unwrap();
+
+    let mut pieces: FlatPieces = vec![0u8; 1024 * PIECE_SIZE].try_into().unwrap();
+    // Generate deterministic pieces, such that we don't have random errors in CI
+    StdRng::seed_from_u64(0).fill(pieces.as_mut());
+
+    let mut piece_indexes = (0..1024).collect::<Vec<_>>();
+    piece_indexes.sort_by_key(|i| xor_distance((*i).into(), address));
+
+    let expected_commitments = piece_indexes[..max_pieces as usize]
+        .iter()
+        .map(|&i| (i, pieces.chunks(PIECE_SIZE).nth(i as usize).unwrap()))
+        .map(|(i, piece)| (i, *<&[u8; PIECE_SIZE]>::try_from(piece).unwrap()))
+        .map(|(i, piece)| (subspace_solving::create_tag(piece, salt), i as PieceIndex))
+        .collect::<BTreeMap<Tag, PieceIndex>>();
+
+    plot.write_many(Arc::new(pieces), 0).unwrap();
+    assert_eq!(plot.piece_count(), max_pieces);
+
+    commitments.create(salt, plot.clone()).unwrap();
+
+    let get_solutions = |target, solution_range, salt| {
+        commitments
+            .find_by_range_many::<BTreeMap<_, PieceOffset>>(target, solution_range, salt)
+            .unwrap()
+            .into_iter()
+            .map(|(tag, offset)| (tag, plot.read_piece_with_index(offset).unwrap().1))
+    };
+
+    // Check all commitments
+    let target = rand::random();
+    let all_commitmens = get_solutions(target, u64::MAX, salt).collect::<BTreeMap<_, _>>();
+    assert_eq!(all_commitmens, expected_commitments);
+
+    // make random queries
+    for _ in 0..10 {
+        let target = rand::random();
+        let range = rand::random();
+
+        let (lower, lower_overflow) = u64::from_be_bytes(target).overflowing_sub(range / 2);
+        let (upper, upper_overflow) = u64::from_be_bytes(target).overflowing_add(range / 2);
+        let expected = if lower_overflow || upper_overflow {
+            expected_commitments
+                .range(..upper.to_be_bytes())
+                .chain(expected_commitments.range(lower.to_be_bytes()..))
+                .map(|(&key, &val)| (key, val))
+                .collect()
+        } else {
+            expected_commitments
+                .range(lower.to_be_bytes()..upper.to_be_bytes())
+                .map(|(&key, &val)| (key, val))
+                .collect::<BTreeMap<_, _>>()
+        };
+
+        let got = get_solutions(target, range, salt).collect::<BTreeMap<_, _>>();
+
+        assert_eq!(expected, got);
+    }
```

Comment:
Should we just use property based testing for this kind of things? Some crates like [this](https://crates.io/crates/quickcheck)

Reply:
Not sure. I looked at examples and it is unclear for me how it is beneficial.